File: 1.24/elixir/solution.exs
Code.require_file("fermat_fast_test.exs", __DIR__)

defmodule TimedFermatPrimeTest do
  def search_for_primes(_, 0), do: nil
  def search_for_primes(number, count) do
    if FermatFastTest.prime?(number) do
      report(number)
      search_for_primes(number + 1, count - 1)
    else
      search_for_primes(number + 1, count)
    end
  end

  def report(number) do
    {time, _result} = :timer.tc(FermatFastTest, :prime?, [number])
    IO.puts "#{number} is prime, time: #{time} mks"
  end
end

Enum.each((0..13), fn(num) ->
  TimedFermatPrimeTest.search_for_primes(1000 * :math.pow(10, num) |> round, 3)
end)

File: 1.24/ruby/solution.rb
require_relative 'timed_fermat_prime_test.rb'

reports = []

10.times do |i|
  TimedFermatPrimeTest.search_for_primes(1000 * (10**i), 3, reports)
end

reports.each do |report|
  puts "Elapsed time for #{report[:number]} is #{report[:time].inspect} ms" 
end

base = 1000
prev = nil

reports.map { |r| r[:time] }.each_slice(3) do |arr|
  avg = arr.sum / arr.size
  str = "#{base} => #{avg} ms"
  str << ": #{avg / prev} times slower" if prev 
  puts str
  prev = avg
  base *= 10
end